blueprint/core/blueprint_CanvasView.h

```diff
@@ -21,6 +21,190 @@ namespace blueprint
+
+                            const int lineWidth = args.arguments[0];
+                            properties.lineWidth = lineWidth;
+
+                            return juce::var();
+                        }
+                });
+            }
+
+            void registerNativeFunctions(juce::Graphics& g)
+            {
+                setProperty("fillRect", juce::var::NativeFunction {
+                        [&g](const juce::var::NativeFunctionArgs& args) -> juce::var {
+                            jassert(args.numArguments == 4);
+
+                            const int x = args.arguments[0];
+                            const int y = args.arguments[1];
+                            const int width = args.arguments[2];
+                            const int height = args.arguments[3];
+
+                            g.fillRect(x, y, width, height);
+
+                            return juce::var();
+                        }
+                });
+
+                setProperty("strokeRect", juce::var::NativeFunction {
+                        [&g, this](const juce::var::NativeFunctionArgs& args) -> juce::var {
+                            jassert(args.numArguments == 4);
+
+                            const int x = args.arguments[0];
+                            const int y = args.arguments[1];
+                            const int width = args.arguments[2];
+                            const int height = args.arguments[3];
+
+                            g.drawRect(x, y, width, height, properties.lineWidth);
+
+                            return juce::var();
+                        }
+                });
+
+                //TODO: Add clearRect. Use View background color and g.fillRect?
+
+                setProperty("strokeRect", juce::var::NativeFunction {
+                        [&g, this](const juce::var::NativeFunctionArgs& args) -> juce::var {
+                            jassert(args.numArguments == 4);
+
+                            const int x = args.arguments[0];
+                            const int y = args.arguments[1];
+                            const int width = args.arguments[2];
+                            const int height = args.arguments[3];
+
+                            g.drawRect(x, y, width, height, properties.lineWidth);
+
+                            return juce::var();
+                        }
+                });
+
+                // Path functions
+                setProperty("beginPath", juce::var::NativeFunction {
+                        [this](const juce::var::NativeFunctionArgs& args) -> juce::var {
+                            jassert(args.numArguments == 0);
+
+                            // Reset the current path on a call to beginPath on the context.
+                            path = juce::Path();
+
+                            return juce::var();
+                        }
+                });
+
+                setProperty("lineTo", juce::var::NativeFunction {
```

Comment:
I need to test this further before anything gets merged. Just pushed to illustrate how some initial Path support might look.